packages/flutter/lib/src/material/chip.dart

```diff
@@ -90,6 +95,11 @@ class Chip extends StatelessWidget {
   /// widget's label.
   final Color backgroundColor;
 
+  /// The border to draw around the chip.
+  ///
+  /// Defaults to [BorderSide.none].
+  final BorderSide border;
```

Comment:
I think rather than a BorderSide I would take a ShapeBorder (and default to `const StadiumBorder(side: const BorderSide.zero)`). Eventually we'll be moving most of our widgets to taking a ShapeBorder, for flexibility.